syntax = "proto3";

package corfureplication;

// Service definition for Corfu replication
service CorfuReplicationService {
  rpc Replicate (CorfuReplicationRequest) returns (CorfuReplicationResponse) {}
}

// Request message containing the data to be replicated
message CorfuReplicationRequest {
  int64 offset = 1;
  int64 size = 2;
  bytes data = 3;
}

// Response message with the result of the replication operation
message CorfuReplicationResponse {
  bool success = 1;
}
